docs/modules/data-structures/pages/working-with-external-data.adoc

```diff
@@ -31,15 +31,88 @@ NOTE: The data store that you choose must be a centralized system that is access
 
 NOTE: The `MapStore` interface extends the `MapLoader` interface. Therefore, all methods and configuration parameters of the `MapLoader` interface are also available on the `MapStore` interface.
 
+== Defining External Data Stores (Beta)
+
+You can define external data stores in your `hazelcast.xml/hazelcast.yaml` configuration files or xref:configuration:configuring-programmatically.adoc[programmatically]. You can then reference these data stores in the xref:integrate:jdbc-connector.adoc[JDBC Connector] and in `MapStores` and `MapLoaders`.
+
+The following example defines a MySQL and a H2 data store.
+[tabs]
+====
+XML::
++
+--
+[source,xml]
+----
+<hazelcast>
+    ...
+    <external-data-store name="my-mysql-database">
+        <class-name>com.hazelcast.datastore.JdbcDataStoreFactory</class-name>
+        <properties>
+            <property name="jdbcUrl">jdbc:mysql://mysql.example.org:3306</property>
+            <property name="username">my_user</property>
+            <property name="password">my_password</property>
+        </properties>
+        <shared>true</shared>
+    </external-data-store>
+    <external-data-store name="my-other-database">
+        <class-name>com.hazelcast.datastore.JdbcDataStoreFactory</class-name>
+        <properties>
+            <property name="jdbcUrl">jdbc:h2:mem:my-other-database</property>
+        </properties>
+        <shared>true</shared>
+    </external-data-store>
+    ...
+</hazelcast>
+----
+--
+
+YAML::
```

Comment:
We should have an example for doing this is Java too.

Reply:
Done: https://github.com/hazelcast/hz-docs/pull/514/commits/b5b20b3d096d78709ff7ef384743b4dc38bbba7f